Source/Schema.NET/ValuesJsonConverter.cs

```diff
@@ -16,7 +16,22 @@ namespace Schema.NET
     /// <seealso cref="JsonConverter" />
     public class ValuesJsonConverter : JsonConverter
     {
-        private const string NamespacePrefix = "Schema.NET.";
+        private static readonly Dictionary<string, TypeInfo> ThingTypeLookup = new Dictionary<string, TypeInfo>();
+
+        static ValuesJsonConverter()
+        {
+            var iThingTypeInfo = typeof(IThing).GetTypeInfo();
+            var thisAssembly = iThingTypeInfo.Assembly;
+
+            foreach (var type in thisAssembly.ExportedTypes)
+            {
+                var typeInfo = type.GetTypeInfo();

```

Comment:
What if someone has a custom type? There are devs who have extended Schema.NET with custom schema types built on top of the ones in this project.

Reply:
Yeah, that is a fair call. The old converter did support any type in the "Schema.Net" namespace so if they added any types in a different assembly to that namespace they would be missed by that type caching.

Reply:
Going through how the old support worked, it is a little interesting. Technically it would only support custom types if the generic arguments to `Values<>` or `OneOrMany<>` listed that type specifically. That is, the custom type wouldn't work if an interface or any parent type was used instead.

There was a loophole (I say loophole as I'm not sure it was intended this way) but if you added your type to the namespace "Schema.NET" and put the assembly name in the "@type" property (eg. "MyFancyType, MyAssembly"), it would load your custom type. I didn't pick that up at first as it just seemed like that code was just for loading built-in types. It does however work as a good security measure by requiring the namespace as the idea of loading ANY type and instantiating it could be a big security problem (some poorly written types might actually do work in their constructor or at least be very allocate-y).

So the code as it stood when you saw it supported the former - if a concrete type is used as a generic argument, it would correctly deserialize to it. To have a concrete type as the generic argument is a bit of a chicken-and-egg scenario as to have a custom property with your concrete type requires a concrete type.

What I've done in 536a53ab is extend support to allow the latter - if a type is in the "Schema.NET" namespace, it will look the type up the old way. I've kept the cached type lookup as a means to mitigate some allocations from string concatenation and likely are still faster lookups from the dictionary (though I might try a benchmark testing that specific theory out).

Reply:
Interesting. I have never tried adding custom types myself but I have seen a few devs mention that they do so, so it would be good to continue to support them.

If we could extend support and lift the limitation on the Schema.NET namespace, I think that would be a nice improvement.

Reply:
Yeah, happy to try and make support better for custom types though want to be careful we don't start instantiating any-old type otherwise it might open up security issues as we are deriving the type from the payload - might need to look into this more.

Reply:
As long we check that types inherit from `Thing`, I don't think there should be any security issues.